### DISCUSSION TITLE
how to extend the behavior of the default component "image" when clicking to open the asset manager

### DISCUSSION THREAD

**Discussion by capoia** [GrapesJS/grapesjs#5063] [Q&A]
Hello, I need to create a component called bs-image that has EXACTLY the same behavior as the default image, that is, when I click on it, it opens the asset manager to select the image and when I select it, I change the src to the chosen image.
Following the documentation it says to use:`extendFn: ['init']`
or even in the worst case` domc.getType('parent-type').model.prototype.init.apply(this, arguments)`, however extendFN seems not do absolutely nothing, and the other returns undefined.
![CleanShot 2023-04-20 at 17 30 55](https://user-images.githubusercontent.com/21159928/233480990-832a5dba-58ea-4860-844a-582a71c2bf49.png)

**Comment by artf** [comment]
The init is an empty function by default so not sure what exactly you're trying to extend, but anyway I think your real issue is `extends`... it should be `extend`
